fenicsxprecice/adapter_core.py

```diff
@@ -160,17 +120,24 @@ def convert_fenicsx_to_precice(fenicsx_function, local_ids):
         raise Exception("Cannot handle data type {}".format(type(fenicsx_function)))
 
     precice_data = []
-
-    if fenicsx_function.function_space().num_sub_spaces() > 0:
-        dims = fenicsx_function.function_space().num_sub_spaces()
-        sampled_data = fenicsx_function.compute_vertex_values().reshape([dims, -1])
-    else:
-        sampled_data = fenicsx_function.compute_vertex_values()
+    # sampled_data = fenicsx_function.x.array  # that works only for 1st order elements where dofs = grid points
+    # TODO begin dirty fix
+    x_mesh = fenicsx_function.function_space.mesh.geometry.x
+    x_dofs = fenicsx_function.function_space.tabulate_dof_coordinates()
+    mask = []  # where dof coordinate == mesh coordinate
+    for i in range(x_dofs.shape[0]):
+        for j in range(x_mesh.shape[0]):
+            if np.allclose(x_dofs[i, :], x_mesh[j, :], 1e-15):
+                mask.append(i)
+                break
+    sampled_data = fenicsx_function.x.array[mask]
+    # end dirty fix
 
     if len(local_ids):
-        if fenicsx_function.function_space().num_sub_spaces() > 0:  # function space is VectorFunctionSpace
-            for lid in local_ids:
-                precice_data.append(sampled_data[:, lid])
+        if fenicsx_function.function_space.num_sub_spaces > 0:  # function space is VectorFunctionSpace
+            raise Exception("Functions from VectorFunctionSpaces are currently not supported.")
+            #for lid in local_ids:  
+               #precice_data.append(sampled_data[lid, :])  # this line raises an IndexError
```

Comment:
Please delete the commented out lines of code. Please also link to https://github.com/precice/fenicsx-adapter/issues/9 in the exception.

Reply:
I took care of this in [9e6510a](https://github.com/precice/fenicsx-adapter/pull/15/commits/9e6510aad549a59cf3e57d5d94173e0bbd647638).